validation/validation.go

```diff
@@ -1,22 +1,179 @@
 package validation
 
+import (
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+	"strconv"
+
+	"github.com/helm/helm/chart"
+	"github.com/helm/helm/log"
+	"github.com/helm/helm/manifest"
+	"gopkg.in/yaml.v2"
+)
+
+// ChartValidation represents a specific instance of validation against a specific directory
+type ChartValidation struct {
+	Path         string
+	Validations  []*Validation
+	Chartfile    *chart.Chartfile
+	Manifests    []*manifest.Manifest
+	ErrorCount   int
+	WarningCount int
+}
+
+const (
+	warningLevel = 1
+	errorLevel   = 2
+)
+
 // Validation - hold messages related to validation of something
 type Validation struct {
-	Errors   []string
-	Warnings []string
+	children  []*Validation
+	path      string
+	validator validator
+	Message   string
+	level     int
+}
+
+//ChartYamlPath - path to Chart.yaml
```

Comment:
[Effective Go](https://golang.org/doc/effective_go.html#commentary) recommends that doc comments be complete sentences, so ideally this would be something like:
```
// ChartYamlPath returns the fully-qualified path of the "Chart.yaml" file.
```
Also I'm not sure if any go tools require a space after the `//` comment start, but I think that's more standard and readable.

(This is quite nitpicky, so feel free to ignore the above.)